src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectDefaultConverter.cs

```diff
@@ -400,7 +400,7 @@ internal override bool OnTryRead(ref Utf8JsonReader reader, Type typeToConvert,
                 onSerialized.OnSerialized();
             }
 
-            value = (T)obj; // unbox
+            _ = (T)obj; // unbox
```

Comment:
What's the purpose of this?  To throw an exception if obj isn't a T?

Reply:
It seems like some odd optimization try in #54709. Removed.

Reply:
> Removed

It still appears to be there.  Did you intend to remove it?